pkg/query/logical/expr_literal.go

```diff
@@ -90,32 +89,26 @@ type int64ArrLiteral struct {
 	arr []int64
 }
 
-func (i *int64ArrLiteral) Compare(tagValue *modelv1.TagValue) (int, bool) {
-	intArray := tagValue.GetIntArray()
-	if intArray == nil {
-		return 0, false
-	}
-	if slices.Equal(i.arr, intArray.Value) {
-		return 0, true
+func (i *int64ArrLiteral) Compare(other LiteralExpr) (int, bool) {
+	if o, ok := other.(*int64ArrLiteral); ok {
+		return 0, slices.Equal(i.arr, o.arr)
 	}
 	return 0, false
 }
 
-func (i *int64ArrLiteral) BelongTo(tagValue *modelv1.TagValue) bool {
-	intValue := tagValue.GetInt()
-	if intValue != nil {
-		return slices.Contains(i.arr, intValue.Value)
+func (i *int64ArrLiteral) BelongTo(other LiteralExpr) bool {
```

Comment:
The semantics of `BelongTo` is not consistent for all `ComparableExpr`,

1. For singled values, it mean if `this` is an element of a slice, or it is equal to the element,
2. For slices, it means if the given element is a subset of `this` slice, which inverses the relationship.

Reply:
yes, it sounds like a bug here.